pkg/sypgp/sypgp_test.go

```diff
@@ -197,6 +199,152 @@ func TestPushPubkey(t *testing.T) {
 	}
 }
 
+func TestEnsureDirPrivate(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", "test-ensure-dir-private")
+	if err != nil {
+		t.Fatalf("Cannot create temporary directory")
+	}
+	defer os.RemoveAll(tmpdir)
+
+	cases := []struct {
+		name        string
+		preFunc     func(string) error
+		entry       string
+		expectError bool
+	}{
+		{
+			name:        "non-existent directory",
+			entry:       "d1",
+			expectError: false,
+		},
+		{
+			name:        "directory exists",
+			entry:       "d2",
+			expectError: false,
+			preFunc: func(d string) error {
+				if err := os.MkdirAll(d, 0777); err != nil {
+					return err
+				}
+				return os.Chmod(d, 0777)
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		testEntry := filepath.Join(tmpdir, tc.entry)
+
+		if tc.preFunc != nil {
+			if err := tc.preFunc(testEntry); err != nil {
+				t.Errorf("Unexpected failure when calling prep function: %+v", err)
+				continue
+			}
+		}
+
+		err := ensureDirPrivate(testEntry)
+		switch {
+		case !tc.expectError && err != nil:
+			t.Errorf("Unexpected failure when calling ensureDirPrivate(%q): %+v", testEntry, err)
+
+		case tc.expectError && err == nil:
+			t.Errorf("Expecting failure when calling ensureDirPrivate(%q), but got none", testEntry)
+
+		case tc.expectError && err != nil:
+			t.Logf("Expecting failure when calling ensureDirPrivate(%q), got: %+v", testEntry, err)
+			continue
+
+		case !tc.expectError && err == nil:
+			// everything ok
+
+			fi, err := os.Stat(testEntry)
+			if err != nil {
+				t.Errorf("Error while examining test directory %q: %+v", testEntry, err)
+			}
+
+			if !fi.IsDir() {
+				t.Errorf("Expecting a directory after calling ensureDirPrivate(%q), found something else", testEntry)
+			}
+
+			if actual, expected := fi.Mode() & ^os.ModeDir, os.FileMode(0700); actual != expected {
+				t.Errorf("Expecting mode %o, got %o after calling ensureDirPrivate(%q)", expected, actual, testEntry)
+			}
+		}
+	}
+}
+
+func TestEnsureFilePrivate(t *testing.T) {
```

Comment:
Could you please add `test.DropPrivilege(t)` and `test.ResetPrivilege(t)`? Thanks

Reply:
different PR